nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/QueryRowsMarkLogic.java

```diff
@@ -24,85 +24,106 @@
+		.required(true)
+		.addValidator(Validator.VALID)
+		.expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+		.build();
+
+	public static final PropertyDescriptor MIMETYPE = new PropertyDescriptor.Builder()
+		.name("MIME Type")
+		.displayName("MIME Type")
+		.description("The MIME type to use when returning the rows")
+		.defaultValue("text/csv")
+		.required(true)
+		.addValidator(Validator.VALID)
+		.expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+		.build();
+
+	protected static final Relationship FAILURE = new Relationship.Builder().name("failure")
+		.description("FlowFiles that were not successfully processed are routed here").build();
+
+	protected static final Relationship SUCCESS = new Relationship.Builder().name("success")
+		.description("A FlowFile is routed here with its content being that exported rows").build();
+		
+	protected static final Relationship ORIGINAL = new Relationship.Builder().name("original")
+        .autoTerminateDefault(true)
+            .description("If this processor receives a FlowFile, it will be routed to this relationship").build();
+
+	@Override
+	public void init(ProcessorInitializationContext context) {
+		List<PropertyDescriptor> list = new ArrayList<>();
+		list.add(DATABASE_CLIENT_SERVICE);
+		list.add(PLAN);
+		list.add(MIMETYPE);
+		properties = Collections.unmodifiableList(list);
+
+		Set<Relationship> set = new HashSet<>();
+		set.add(FAILURE);
+		set.add(SUCCESS);
+		set.add(ORIGINAL);
+		relationships = Collections.unmodifiableSet(set);
+	}
+
+	@Override
+	public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFactory) throws ProcessException {
+		final ProcessSession session = sessionFactory.createSession();
+		FlowFile flowFile = session.get();
+		FlowFile incomingFlowFile = flowFile;
+		try {
+			
+			if (flowFile == null) {
+				flowFile = session.create();
+			}
+
+			final String jsonPlan = determineJsonPlan(context, flowFile);
+			final String mimeType = determineMimeType(context, flowFile);
+
+			session.putAttribute(flowFile, "marklogic-optic-plan", determineJsonPlan(context, flowFile));
+
+			final DatabaseClient client = getDatabaseClient(context);
+			final RowManager rowManager = client.newRowManager();
+			PlanBuilder.Plan plan = rowManager.newRawPlanDefinition(new StringHandle(jsonPlan));
+
+			InputStream inputStream = rowManager.resultDoc(plan, new InputStreamHandle().withMimetype(mimeType)).get();
+			if (inputStream != null) {
+				try {
+					if(incomingFlowFile != null) {
+						incomingFlowFile = session.write(flowFile, out -> {
+							FileCopyUtils.copy(inputStream, out);
+						});
+					} else {
+						flowFile = session.write(flowFile, out -> {
+							FileCopyUtils.copy(inputStream, out);
+						});
+					}
+				} finally {
+					inputStream.close();
+				}
+			}
+			if(incomingFlowFile != null) {
```

Comment:
This isn't quite right - we want the `incomingFlowFile` to go to `ORIGINAL` . We then want to clone a new FlowFile from `incomingFlowFile` and write the results to it and send that to `SUCCESS`.